vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -118,6 +118,33 @@ class OnboardingViewModel @AssistedInject constructor(
         }
     }
 
+    private fun observeQrCodeLoginCapability(homeServerUrl: String) = viewModelScope.launch {
+        if (!vectorFeatures.isQrCodeLoginEnabled()) {
+            setState {
+                copy(
+                        canLoginWithQrCode = false
+                )
+            }
+        } else if (vectorFeatures.isQrCodeLoginForAllServers()) {
+            // allow for all servers
+            setState {
+                copy(
+                        canLoginWithQrCode = true
+                )
+            }
+        } else {
+            // check if selected server supports MSC3882 first
+            homeServerConnectionConfigFactory.create(homeServerUrl)?.let {
```

Comment:
Maybe use the block `viewModelScope.launch {` only here?

Reply:
Right, done.